src/core/cm/imagemanager/imagemanager.cpp

```diff
@@ -266,10 +289,52 @@ Error ImageManager::GetUpdateImageInfo(
     return ErrorEnum::eNotFound;
 }
 
-Error ImageManager::GetLayerImageInfo(
-    [[maybe_unused]] const String& digest, [[maybe_unused]] smcontroller::UpdateImageInfo& info)
+Error ImageManager::GetLayerImageInfo(const String& digest, smcontroller::UpdateImageInfo& info)
 {
-    return ErrorEnum::eNone;
+    LockGuard lock {mMutex};
+
+    LOG_DBG() << "Get layer image info" << Log::Field("digest", digest);
+
+    auto items = MakeUnique<StaticArray<storage::ItemInfo, cMaxNumUpdateItems>>(&mAllocator);
+
+    if (auto err = mStorage->GetItemsInfo(*items); !err.IsNone()) {
```

Comment:
Looks like we may have all items cached, instead of reading all items each time.